src/modeler/AutoRest.Swagger/Validation/SkuModelValidation.cs

```diff
@@ -50,7 +60,7 @@ public override bool IsValid(Dictionary<string, Schema> definitions)
 
                     bool hasName = schema.Properties.Any(property =>
                         property.Key.EqualsIgnoreCase("name") &&
-                        property.Value.Type == Model.DataType.String);
+                        (property.Value.Type == Model.DataType.String || (property.Value.Type == null && evaluateIf(property.Value.Reference, definitions, Model.DataType.String))));
```

Comment:
I have renamed the method to IsReferenceOfType. But not after the '||'. Consider a situation such as I actually have the type set to boolean and ref type set to string. In theory, it is possible.